backend/routes/clips.go

```diff
@@ -125,33 +125,40 @@ func (r *Routes) GetClip(user *models.User, req *http.Request) (int, []byte, err
 // GetClipProgress returns the progress of a clip, if it's being processed
 // Reports -1 if the processing has not yet begun, or a number between 0 and 100 based on the progress
 // Returns 204 if the clip is not being processed, or is done processing
-func (r *Routes) GetClipProgress(user *models.User, req *http.Request) (int, []byte, error) {
-	vars := vars(req)
+func (r *Routes) GetProgress(user *models.User, req *http.Request) (int, []byte, error) {
+	queryparams := query(req)
 
-	clip, err := r.Clips.Find(req.Context(), vars.CID)
+	clip, err := r.Clips.FindMany(req.Context(), models.ClipWhere.ID.IN(queryparams.CID))
 
-	if err == sql.ErrNoRows {
-		return http.StatusNotFound, nil, nil
-	} else if err != nil {
+	if err != nil {
 		return http.StatusInternalServerError, nil, err
 	}
 
-	// If either the clip is already done, or it's not being processed, return no content
-	if !clip.Processing {
-		return http.StatusNoContent, nil, nil
+	if len(clip) == 0 {
+		return http.StatusNotFound, nil, nil
 	}
 
-	progress, ok := r.Transcoder.GetProgress(clip.ID)
+	prog := &modelsx.Progress{
+		Clips: make(map[modelsx.HashID]int),
+	}
 
-	if !ok {
-		return http.StatusNoContent, nil, nil
+	for _, c := range clip {
+		if c.Processing {
+			progress, ok := r.Transcoder.GetProgress(c.ID)
+
+			if !ok {
+				continue
```

Comment:
[nit] Once we have Sentry would it be good to log here? Maybe add a TODO so we don't forget this

Reply:
GetProgress is expected to not return anything if the clip is no longer being processed, so this isn't really a fail state.